src/runtime.js

```diff
@@ -510,7 +512,7 @@ function registerServiceForDoc(ampdoc, name, opt_ctor, opt_factory) {
   dev().assert((opt_ctor || opt_factory) && (!opt_ctor || !opt_factory),
       'Only one: a class or a factory must be specified');
   if (opt_ctor) {
-    fromClassForDoc(ampdoc, name, opt_ctor);
+    registerServiceForDocExternal(ampdoc, name, opt_ctor);
```

Comment:
What about services that specify a factory (else branch)? Are you addressing that in a follow-up?